app/src/main/java/io/github/hidroh/materialistic/CustomTabsDelegate.java

```diff
@@ -110,8 +112,36 @@ void onServiceDisconnected() {
     }
 
     private static String getPackageNameToUse(Context context) {
-        // packagesSupportingCustomTabs contains all apps that can handle both VIEW intents
-        // and service calls.
+        List<String> browserWithCustomTabsSupport = getBrowserWithCustomTabsSupport(context);
+        String defaultBrowser = getDefaultBrowser(context);
+
+        for (String browser : browserWithCustomTabsSupport) {
+            if (browser.equals(defaultBrowser)) {
+                return browser;
+            }
+        }
+
+        return browserWithCustomTabsSupport.get(0);
+    }
+
+    /**
+     * Returns the package name of the default browser on the device
+     * @param context Context
+     * @return The package name of the default browser
+     */
+    private static String getDefaultBrowser(Context context) {
+        Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("http://example.com/"));
+        PackageManager pm = context.getPackageManager();
+        ResolveInfo resolveInfo = pm.resolveActivity(browserIntent,PackageManager.MATCH_DEFAULT_ONLY);
```

Comment:
It is possible for this to return null if no matching Activity is found, e.g. if one has no browsers install or no browsers set as default (not sure if this will happen). In such case we should do a null check and return null.

Also can you put a space after `,`? Thanks!

Reply:
>  e.g. if one has no browsers install or no browsers set as default (not sure if this will happen).

On my device a generic package `com.huawei.android.internal.app` is returned if no default browser is set. This might be vendor/device specific.